.github/copilot-instructions.md

```diff
@@ -0,0 +1,212 @@
+
+#### 3. Running Tests and Checks (Essential Commands - validated and working)
+```bash
+./gradlew check
+```
+**Note**: The project has "limited number of tests" according to CONTRIBUTING.md, but this validates what exists.
+
+#### 4. Building Demo Applications
+```bash
+# Build the side-by-side demo (used for GitHub Pages)
+./gradlew :compose-multiplatform-html-unified-demo:sideBySideBrowserDistribution
+
+# Individual platform builds
+./gradlew jsBrowserDistribution    # JS DOM version
+./gradlew wasmJsBrowserDistribution  # Wasm-JS Canvas version
+```
+
+### Build Configuration Details
+
+#### Memory Requirements
+- **Gradle JVM**: 2GB memory allocated via `org.gradle.jvmargs=-Xmx2G` in gradle.properties (required for Wasm compilation)
+
+#### Common Build Issues and Workarounds
+
+1. **Network Connectivity**: Initial builds require internet access for Android Gradle Plugin (8.11.2) and other dependencies. If build fails with network errors, ensure internet connectivity.
+
+2. **Memory Issues**: If Wasm compilation fails with OOM, the memory is already optimized in gradle.properties.
+
+3. **Kotlin/JS Store**: The `kotlin-js-store` directory may be generated during JS builds - this should not be committed.
+
+4. **Gradle Daemon**: May timeout on first runs. Use `--no-daemon` flag if needed: `./gradlew check --no-daemon`
+
+### Build Timing and Known Issues
+
+**Timing Expectations**:
+- First build: 5-10 minutes (includes dependency resolution and Wasm compilation)
+- Subsequent builds: 1-3 minutes
+- `publishToMavenLocal`: 3-5 minutes
+- `check` execution: 2-5 minutes (limited tests)
+- Demo builds: 2-4 minutes
+
+**Common Issues**:
+- **Wasm Compilation**: Requires 2GB+ heap memory (pre-configured in gradle.properties)
+- **Network Dependencies**: Initial builds require internet access for dependency resolution
+- **Platform-specific builds**: Some targets may be disabled on certain OS (use `--continue` flag)
+
+## Project Architecture and Layout
+
+### Module Structure
+```
+├── common/                    # Core APIs, layouts, modifiers (foundation equivalent)
+├── material-icons-core/       # Icon system core
+├── material2/                 # Material Design 2 components  
+├── material3/                 # Material Design 3 components
+├── navigation/               # Navigation support (experimental)
+├── lifecycle-viewmodel/      # ViewModel support (experimental)
+├── demo/                     # Demonstration application
+└── buildSrc/                 # Build configuration and dependencies
+```
+
+### Key Configuration Files
+- **buildSrc/src/main/kotlin/VersionsAndDependencies.kt**: All dependency versions
+- **buildSrc/src/main/kotlin/common-conventions.gradle.kts**: Shared build configuration
+- **buildSrc/src/main/kotlin/lib-conventions.gradle.kts**: Library-specific configuration
+- **gradle.properties**: JVM memory settings and Kotlin MPP configuration
```

Comment:
As above, is it necessary to mention the contents of `gradle.properties` here? The contents may be constantly updated. Maybe you can just mention that it includes JVM memory settings so that you know what to increase if there is an OOM from Gradle.